### ISSUE TITLE
[FIXED] The first DocumentTab to be opened appears as a small square in upper left before expanding to fill the MDICLIENT

### ISSUE THREAD

**Issue by astrohart** [dockpanelsuite/dockpanelsuite#372]
BTW -- I have a fix for this issue already in hand -- so I am merely posting this to (a) document the bug, and (b) point out which code needs to be fixed.

So, I have a Visual Studio-like application and it starts out with a DockWindow docked to the left (i.e., DockState==DockLeft).  When I open the first DockingMdi MDI Document tab, the tab appears first as a small square of gray covering the dock window and then it expands to fill the space left over to the right.  

This is done even when the Form.Load of the enclosed DockContent does not process a lengthy operation.  It causes a noticeable flicker.  This only happens the first time a Document is opened; all other documents from then on appear automatically to be the correct size.

Here is what I mean: I open up my app and then create a File -> New document.  First, we see the following:

![pic1](https://cloud.githubusercontent.com/assets/8261875/17745706/c14f5afa-6461-11e6-8bab-341ec1659618.png)

Then, a few seconds later, the form fills the proper amount of the screen real estate:

![pic2](https://cloud.githubusercontent.com/assets/8261875/17745730/d9a0a49c-6461-11e6-9a8e-16f0bd1f6dc4.png)

The desired behavior is for the document tab not to appear until it's the correct size.  I know the solution.  In the original codebase, the DockWindow class overrides the OnLayout method:

// original code
protected override void OnLayout(LayoutEventArgs levent)
{
    VisibleNestedPanes.Refresh();
    if (VisibleNestedPanes.Count == 0)
    {
        if (Visible)
            Visible = false;
    }
    else if (!Visible)
    {
        Visible = true;
        VisibleNestedPanes.Refresh();
    }

    base.OnLayout(levent);
}

I am aware that this is probably not the root cause of the issue, but doing the following code change appears to fix the issue (i.e., the application's first document tab is not displayed until the content window has already had its bounds refreshed to fill the space to the right of the dock-left DockWindow):

// fixed code
protected override void OnLayout(LayoutEventArgs levent)
{
    VisibleNestedPanes.Refresh();
    if (VisibleNestedPanes.Count == 0)
    {
        if (Visible)
            Visible = false;
    }
    else if (!Visible)
    {
        VisibleNestedPanes.Refresh();
        Visible = this.DockState != DockState.Document;
    }

    base.OnLayout(levent);

    if (this.DockState != DockState.Document
        || !"Bounds".Equals(levent.AffectedProperty))
    {
        return;
    }

    Visible = true;
} 

Basically, the idea is, go ahead and set Visible = true when necessary for anything but a DockState.Document.  Then, call the base class' OnLayout method (which is the ScrollableControl version) and then, test for whether the DockState == Document AND "Bounds" is the value of levent.AffectedProperty.  If so, then it's time to set Visible = true; otherwise, we are done with the method.

Like I said, I realize I may have not fixed the actual root cause of this issue, but I believe it to be well-patched.

**Comment by astrohart** [comment]
As the fix has been documented along with the issue, I have closed it; however, I strongly recommend that this update be put into the master branch.

**Comment by roken** [comment]
Please feel free to submit a pull request with the fix for inclusion into master.

**Comment by lextm** [comment]
With the new mechanism of #369 we can even ship your patch to all users (with a switch to disable it if anything does not work out). So any patch is now welcome.

You can now send a normal PR where I can help put it under patch controller afterwards, or you can check out commits such as 20be7515575d0a1940dd3c0bd1380c2849662665 to integrate with patch controller on your own.
